docs/core/baseapp.md

```diff
@@ -0,0 +1,329 @@
+
+Before the first transaction of a given block is processed, a [volatile state](#volatile-states) called `deliverState` is intialized during [`BeginBlock`](#beginblock). This state is updated each time a transaction is processed via `DeliverTx`, and committed to the [main state](#main-state) when the block is [committed](#commit), after what is is set to `nil`. 
+
+`DeliverTx` performs the **exact same steps as `CheckTx`**, with a little caveat at step 3 and the addition of a fifth step:
+
+3. The `anteHandler` does **not** check that the transaction's `gas-prices` is sufficient. That is because the `min-gas-prices` value `gas-prices` is checked against is local to the node, and therefore what is enough for one full-node might not be for another. This means that the proposer can potentially include transactions for free, although they are not incentivised to do so, as they earn a bonus on the total fee of the block they propose. 
+5. For each `message` in the transaction, route to the appropriate module's [`handler`](../building-modules/handler.md). Additional *stateful* checks are performed, and the cache-wrapped multistore held in `deliverState`'s `context` is updated by the module's `keeper`. If the `handler` returns successfully, the cache-wrapped multistore held in `context` is written to `deliverState` `CacheMultiStore`. 
+
+During step 5., each read/write to the store increases the value of `GasConsumed`. You can find the default cost of each operation [here](https://github.com/cosmos/cosmos-sdk/blob/master/store/types/gas.go#L142-L150). At any point, if `GasConsumed > GasWanted`, the function returns with `Code != 0` and `DeliverTx` fails. 
+
+`DeliverTx` returns a response to the underlying consensus engine of type [`abci.ResponseCheckTx`](https://tendermint.com/docs/spec/abci/abci.html#messages). The response contains:
+
+- `Code (uint32)`: Response Code. `0` if successful. 
+- `Data ([]byte)`: Result bytes, if any.
+- `Log (string):` The output of the application's logger. May be non-deterministic.
+- `Info (string):` Additional information. May be non-deterministic.
+- `GasWanted (int64)`: Amount of gas requested for transaction. It is provided by users when they generate the transaction. 
+- `GasUsed (int64)`: Amount of gas consumed by transaction. During `DeliverTx`, this value is computed by multiplying the standard cost of a transaction byte by the size of the raw transaction (click [here](https://github.com/cosmos/cosmos-sdk/blob/master/x/auth/ante.go#L101) for an example), and by adding gas each time a read/write to the store occurs. 
+- `Tags ([]cmn.KVPair)`: Key-Value tags for filtering and indexing transactions (eg. by account).
+- `Codespace (string)`: Namespace for the Code.
+
+## RunTx, AnteHandler and RunMsgs
+
+### RunTx
+
```

Comment:
I think it would be more clear with a proper rendering of the page. If you look at the sectioning, you'll see that `## Run Tx, AnteHandler and RunMsgs` is not under `Main ABCI Messages`. These method are called from methods that implement the ABCI though, and we need to explain what they do.